test fails with only feature dangerous_configuration

One test fenced for feature `dangerous_configuration` also requires feature `tls12`:

https://github.com/rustls/rustls/blob/c85651398865a44fe163003c3eec79a3af914362/rustls/tests/server_cert_verifier.rs#L70

Otherwise it fails like this:
```
error[E0425]: cannot find value `TLS12` in module `rustls::version`
  --> tests/server_cert_verifier.rs:70:91
   |
70 |           let mut client_config = make_client_config_with_versions(*kt, &[&rustls::version::TLS12]);
   |                                                                                             ^^^^^ help: a stati
c with a similar name exists: `TLS13`
```

--- a/rustls/src/client/tls12.rs
+++ b/rustls/src/client/tls12.rs
@@ -992,6 +992,7 @@ impl ExpectFinished {
 
         let time_now = match TimeBase::now() {
             Ok(time_now) => time_now,
+            #[allow(unused_variables)]
             Err(e) => {
                 debug!("Session not saved: {}", e);
                 return;

--- a/rustls/src/server/hs.rs
+++ b/rustls/src/server/hs.rs
@@ -1,5 +1,7 @@
 use crate::conn::{CommonState, ConnectionRandoms, State};
-use crate::enums::{CipherSuite, ProtocolVersion, SignatureScheme};
+#[cfg(feature = "tls12")]
+use crate::enums::CipherSuite;
+use crate::enums::{ProtocolVersion, SignatureScheme};
 use crate::error::Error;
 use crate::hash_hs::{HandshakeHash, HandshakeHashBuffer};
 #[cfg(feature = "logging")]

--- a/rustls/tests/api.rs
+++ b/rustls/tests/api.rs
@@ -3205,6 +3205,7 @@ mod test_quic {
         }
     }
 
+    #[cfg(feature = "tls12")]
     #[test]
     fn test_quic_no_tls13_error() {
         let mut client_config =

--- a/rustls/tests/server_cert_verifier.rs
+++ b/rustls/tests/server_cert_verifier.rs
@@ -64,6 +64,7 @@ fn client_can_override_certificate_verification_and_reject_certificate() {
     }
 }
 
+#[cfg(feature = "tls12")]
 #[test]
 fn client_can_override_certificate_verification_and_reject_tls12_signatures() {
     for kt in ALL_KEY_TYPES.iter() {

